SOURCES/etc/xapi.d/plugins/updater.py

```diff
@@ -43,9 +106,10 @@ def check_update(session, args):
     return json.dumps(map(display_package, packages))
 
 
+@operationlock(timeout=10)
```

Comment:
I'm not entirely sure, but it looks to me that in the following scenario, we'd wait for 10s instead of raising immediately:
* `self.operation_progress == 'update'`
* `self.operation == 'update'`
In this situation it's not useful to keep retrying for 10s. We should fail fast and only keep retrying when we're waiting for a check_update.
Am I right?

Reply:
https://github.com/xcp-ng/xcp-ng-xapi-plugins/blob/5cbf2e06a6dacddc4e7ce549db77dde9a168fa8d/SOURCES/etc/xapi.d/plugins/xcpngutils/filelocker.py#L50-L58
https://github.com/xcp-ng/xcp-ng-xapi-plugins/blob/5cbf2e06a6dacddc4e7ce549db77dde9a168fa8d/SOURCES/etc/xapi.d/plugins/updater.py#L39-54 `_prelock` is called before the try lock phase. At this moment the current operation is read from lock file and compared to the current operation. So there is no issue concerning the timeout here, the comparison is immediate before the timeout.